### PR TITLE
Ins 3589 disallow side effect on outgoing

### PR DESCRIPTION
**Pull request by SergeyRadist**
**- What I did**
added new check in set_result that disallows side effect on outgoing request
**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger/light/proc/set_result.go [hunk 1]
```diff
@@ -148,6 +148,10 @@ func (p *SetResult) Proceed(ctx context.Context) error {
 		return errors.Wrap(err, "failed to calculate pending requests")
 	}
 	closedRequest, err := findClosed(opened, p.result)
+
+	if p.sideEffect != nil {
+		err = checkRequest(closedRequest)
```
Comment (bigbes) [inline ledger/light/proc/set_result.go]: checkRequestSideEffect?

File: insolar/payload/error_code.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ const (
 	CodeReasonNotFound    = 1008
 	CodeReasonIsWrong     = 1009
 	CodeNonClosedOutgoing = 1010
+
+	CodeRequestIsWrong = 1011
```
Comment (andreyromancev) [inline insolar/payload/error_code.go]: We already have `CodeInvalidRequest`

File: insolar/record/record.go [hunk 3]
```diff
@@ -164,6 +164,8 @@ type Request interface {
 	// IsTemporaryUploadCode tells us that that request is temporary hack
 	// for uploading code.
 	IsTemporaryUploadCode() bool
+	// IsMutable returns mutable-state of request
+	IsMutable() bool
```
Comment (andreyromancev) [inline insolar/record/record.go]: This method is not necessary. Mutable is IncomingRequest property.

File: ledger/light/proc/set_result.go (hunk 1 above)
Comment (SergeyRadist) [reply ledger/light/proc/set_result.go]: What about checkRequestCanChangeState ?

File: insolar/record/record.go (hunk 3 above)
Comment (SergeyRadist) [reply insolar/record/record.go]: reverted

**Pull request merged**